src/features/create-quiz/store/CreateQuizStore.tsx

```diff
@@ -31,37 +29,103 @@ export class QuizStoreImpl {
     this.questions.push(item);
   }
 
-  editQuestion(questionId: string, updatedQuestion: Question | null) {
+  editQuestion(questionId: string, updatedQuestion: IQuestion | null) {
     const result = this.questions.find(
       (question) => question.id === questionId
     );
     if (result) {
       result.question = updatedQuestion;
-      this.selectedQuestionID = "";
+      this.selectQuestion(this.questions[this.questions.length - 1]);
     }
   }
 
   removeQuestion(id: string) {
     this.questions = this.questions.filter((Question) => Question.id !== id);
   }
 
+  clearQuestions() {
+    this.questions = [];
+  }
+
   toggleQuestion(id: string) {
     const index = this.questions.findIndex((item) => item.id === id);
     if (index > -1) {
       this.questions[index].completed = !this.questions[index].completed;
     }
   }
 
-  selectQuestion(question: QuestionItem | null) {
+  setFetchedFirebaseQuizzes(fetchedQuizzes: IQuizItem[]) {
```

Comment:
I think the main purpose of this function is just to set the quizzes array regardless of where they come from, so a more generic naming for the function might be just `setQuizzes`.

Reply:
And for the parameter - `quizzes`